src/sqlfluff/core/templaters/base.py

```diff
@@ -22,6 +23,32 @@ def iter_indices_of_newlines(raw_str: str) -> Iterator[int]:
             break  # pragma: no cover TODO?
 
 
+def large_file_check(func):
+    """Raise an exception if the file is over a defined size.
+
+    Designed to be implemented as a decorator on `.process()` methods.
+
+    If no config is provided or the relevant config value is set
+    to zero then the check is skipped.
+    """
+
+    def _wrapped(
+        self, *, in_str: str, fname: str, config: FluffConfig = None, **kwargs
```

Comment:
Since the parameter list is closely tied to the templaters' `process()` function, what would you think of moving the check to the templater base class function, `RawTemplater.process()`? We could have each templater's `process()` function call the base class `process()` function before doing any work.

That seems a bit cleaner to me (because decorators are usually somewhat generic in terms of the functions they can wrap).

Reply:
That approach is definitely cleaner. It feels like a big change - but maybe you're right and I should just bit the bullet on that. It probably triggers a bigger version change.

Reply:
I don't think it'll be bad -- just adding a call to `super().process()` in the subclasses. Dbt could be a bit different, if it inherits from Jinja templater.

Let me know if I can help with tactical advice or whatever!

Reply:
Another option: add a `large_file_check()` function to the base class, then each process() function calls it. A bit simpler than calling the base class process() function.

Reply:
Actually on second thoughts I'm not sure. Here are the options I see:

1. Decorator - as per what I've done.
2. Simple function call (probably a method of the base class like `.do_large_file_check()`) which each `.process()` method calls before doing anything else. I did this first and it felt messy. Defined once yes, but feels a little fragile. Simpler than option 1 though.
3. Put it in the `RawTemplater.process()` method and then call `.super()` in other process methods. This gets messy because the dbt templater inherits from then jinja templater, which inherits from the python templater, which inherits from the RawTemplater. We wouldn't just want to call `super()` there - we'd be referencing `super()` in some cases and `RawTemplater` in others. This just feels like it just ends up as a more complicated version of option 2.
4. Have two methods on templaters one which calls the other. The outer is only defined on the `RawTemplater`, and does the large file checks, then calls the inner. The inner is the one defined on all the other templaters.
   i. Keeping the inner method as `.process()` and calling the outer something else changes the API which the templaters expose to the linter and other internal tooling.
   ii. Keeping the outer method as `.process()` which then calls a renamed inner method means none of the outer APIs change but all the templaters need updating.

I'm not sure any of these options is meaningfully cleaner than another. @barrywhart - views?

Reply:
The simplest option actually feels like option 1 🤷 . Dumb, but simple and easy.

Reply:
Yep, let's go with it. I pride myself on being an "it depends" kind of person. Thanks for thinking it through!